test/e2e/validation_test.go

```diff
@@ -24,6 +24,62 @@ import (
 	nc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachine/nodeconfig"
 )
 
+// testNodeMetadata tests if all nodes have a worker label and kubelet version and are annotated with the version of
+// the currently deployed WMCO
+func testNodeMetadata(t *testing.T) {
+
+	// The worker label is not actually added by WMCO however we would like to validate if the Machine Api is properly
+	// adding the worker label, if it was specified in the MachineSet. The MachineSet created in the test suite has
+	// the worker label
+	workerLabelTest := "Worker Label"
+	kubeletVersionTest := "Kubelet Version"
+	versionAnnotationTest := "Version Annotation"
+
+	operatorVersion, err := getWMCOVersion()
+	require.NoError(t, err, "could not get WMCO version")
+
+	testCases := []struct {
+		testCase string
+		check    string
+	}{
+		{
+			testCase: workerLabelTest,
+			check:    nc.WorkerLabel,
+		},
+		{
+			testCase: kubeletVersionTest,
+			check:    "v0.0.0",
```

Comment:
Please add a comment here saying `v0.0.0-master+$Format:%h$` is an example of a mis-reported version. Though ideally you know the kube version of the cluster. Shouldn't you be asserting that the expected version is present?